src/matlab2tikz.m

```diff
@@ -892,26 +892,52 @@ case guitypes()
 end
 % ==============================================================================
 function m2t = drawGridOfAxes(m2t, handle)
-    % draws the grids of an axes
-    %TODO: has{XYZ}Grid is always false without a good reason
-    hasXGrid = false;
-    hasYGrid = false;
-    hasZGrid = false;
-    if hasXGrid || hasYGrid || hasZGrid
-        matlabGridLineStyle = get(handle, 'GridLineStyle');
-        % Take over the grid line style in any case when in strict mode.
-        % If not, don't add anything in case of default line grid line style
-        % and effectively take Pgfplots' default.
-        defaultMatlabGridLineStyle = ':';
-        if m2t.cmdOpts.Results.strict ...
-                || ~strcmpi(matlabGridLineStyle,defaultMatlabGridLineStyle)
-            gls = translateLineStyle(matlabGridLineStyle);
+    % Draws the grids of an axis
+    % Check for major grids
+    hasGrid =  isOn(get(handle, 'XGrid')) ...
+            || isOn(get(handle, 'YGrid')) ...
+            || isOn(get(handle, 'ZGrid'));
+
+    % Check for minor grids
+    hasMinorGrid =  isOn(get(handle, 'XMinorGrid')) ...
+                 || isOn(get(handle, 'YMinorGrid')) ...
+                 || isOn(get(handle, 'ZMinorGrid'));
+
+    % Default matlab grid style ommited for brevity
+    % defaultGridLineStyle = ':';
```

Comment:
So i commented it out, as the length of defaultGridLineStyle is rather lengthy in comparison with ':'